flexprep/domain/processing.py

```diff
@@ -122,20 +121,40 @@ def _save_output(
         forecast_ref_time_str = forecast_ref_time.strftime("%Y%m%d%H%M")
 
         try:
-            key = f"output_dispf{forecast_ref_time_str}_step{step_to_process}"
+            key = f"output_dispf{forecast_ref_time_str}{step_to_process}"
+
+            ref = next(
+                f
+                for f in ds_out.values()
+                if metadata.extract_keys(f.message, "editionNumber") == 2
+            )
 
             with tempfile.NamedTemporaryFile(
-                suffix=key,
+                suffix=key, delete=False, mode="wb"
             ) as output_file:
-                # Write data to the temporary file
-                with open(output_file.name, "wb") as fout:
-                    for name, field in ds_out.items():
-                        if field.attrs.get("v_coord") == "hybrid":
-                            logger.info(f"Writing GRIB fields to {output_file.name}")
-                            grib_decoder.save(field, fout)
-
-                    # Upload the file to S3
-                    S3client().upload_file(output_file.name, key=key)
+                for name, field in ds_out.items():
+                    if field.squeeze().isnull().all():
```

Comment:
Why squeeze?

Reply:
I wanted to be sure that the check is not done on axis separately.